ztp/extra-manifests-builder/01-disk-encryption-pcr-rebind/luks-helpers.sh

```diff
@@ -0,0 +1,159 @@
+DEBUG="true"
+RESERVED_SLOT=31
+CLEVIS_CONFIG_RESERVED_SLOT="$RESERVED_SLOT: tpm2 '{\"hash\":\"sha256\",\"key\":\"ecc\"}'"
+TRUE=0
+FALSE=1
+NEWLINE=$'\n'
+
+#set -x
+
+# log function. Takes 2 arguments:
+# log level: debug or info
+# string to print
+log() {
+    case $1 in
+    "debug")
+        echo "DEBUG - $2" >&2
+        ;;
+    "info")
+        echo "INFO - $2" >&2
+        ;;
+    *)
+        # Code to execute when no patterns match
+        ;;
+    esac
+}
+
+# logs a string with a debug level
+logDebug() {
+    if ! [ -v DEBUG ] || ([ -v DEBUG ] && [ "$DEBUG" == "true" ]); then
+        log "debug" "$1"
+    fi
+}
+
+# logs a string with a info level
+logInfo() {
+    log "info" "$1"
+}
+
+# return $TRUE id the temporary reserved slot is configured with a key (to disable PCR protection), returns $FALSE otherwise
+isReservedSlotPresent() {
+    RESULT=$(clevis luks list -d $1 -s $RESERVED_SLOT || true)
+    if [ -n "$RESULT" ] && [ "$RESULT" == "$CLEVIS_CONFIG_RESERVED_SLOT" ]; then
+        logDebug "reserved slot $RESERVED_SLOT is present"
+        return $TRUE
+    fi
+    logDebug "reserved slot $RESERVED_SLOT is not present"
+    return $FALSE
+}
+
+# create a temporary key in the reserved slot to disable PCR protection
+addReservedSlot() {
+    logInfo "reservedSlotPresent=$1 device=$2 slot=$3 with PCR IDs=$4 and clevis config=$5"
+    if [ $1 == $TRUE ]; then
+        logInfo "reserve slot already present, no need to add again"
+        clevis luks list -d $2 || true
+        return
+    fi
+    logInfo "adding reserved slot on device=$device"
+    ANYPASS=$(openssl rand -base64 21)
+    echo -e "$ANYPASS\n" | clevis luks bind -s $RESERVED_SLOT -d $2 tpm2 '{}' || true
+    clevis luks list -d $2 || true
+}
+
+# remove the temporary key in the reserved slot to enable PCR protection
+removeReservedSlot() {
+    logInfo "removing luks reserved slot 31 in disk $1"
+    echo "clevis luks unbind -s $RESERVED_SLOT -d $1 -f" | sed 's@/@\/@g' | bash || true
```

Comment:
can you rewrite it without the sed?

Reply:
maybe 
`-d $1` -> `-d \"$1\"`

Reply:
if not, maybe worth adding a comment it's because of clevis oddity

Reply:
fixed